README.md

```diff
@@ -273,7 +282,107 @@ should mention how to instantiate those variables/files, etc.
 
 ### Testing
 
-(TODO: Write me)
+
+There are many ways to test your code. Remember when you ran your application and used it for the first time? Did you check the features and 
+experiment using them? That’s known as exploratory testing and is a form of manual testing.
+
+Exploratory testing is a form of testing that is done without a plan. In an exploratory test, you’re just exploring the application.
+
+To have a complete set of manual tests, all you need to do is make a list of all the features your application has, the different types of input it can accept, 
+and the expected results. Now, every time you make a change to your code, you need to go through every single item on that list and check it.
+
+That doesn’t sound like much fun, does it?
+
+This is where automated testing comes in. Automated testing is the execution of your test plan (the parts of your application you want to test, 
+the order in which you want to test them, and the expected responses) by a script instead of a human. Software testing involves the execution of a 
+software component or system component to evaluate one or more properties of interes. In general, these properties indicate the extent to which the component
+ or system under test:
+
+* meets the requirements that guided its design and development,
+* responds correctly to all kinds of inputs,
+* performs its functions within an acceptable time,
+* is sufficiently usable,
+* can be installed and run in its intended environments, and
+* achieves the general result its stakeholders desire.
+
+#### Unit Tests vs. Integration Tests
+
+What is the Unit Test?
+Unit Tests are conducted by developers and test the unit of code( aka module, component) he or she developed. 
+It is a testing method by which individual units of source code are tested to determine if they are ready to use. 
+It helps to reduce the cost of bug fixes since the bugs are identified during the early phases of the development lifecycle.
+
+What is Integration Test?
+Integration testing tests integration between software modules. It is a software testing technique where individual units of a program 
+are combined and tested as a group. It checks the overall flow of the application after the integration of different modules.
+
+
+
+#### Test driven development
+Test-driven development (TDD) is a software development process that relies on the repetition of a very short development cycle: requirements are turned into 
+very specific test cases, then the code is improved so that the tests pass. In simple terms, 
+test cases for each functionality are created and tested first and if the test fails then the new code is written in order to pass the test 
+and making code simple and bug-free.
+
+
+Useful links for writing tests in our most common  languages and frameworks: 
```

Comment:
These links are not related to Test-driven development which this section is about.

Reply:
I moved the CI/CD  section to Testing. In that way it does not have so much weight in the overall content of the dev guidelines. You ask me:
Are these practices we implement? strive toward? are inspired by? ignore? The sections do not say.
I guess we have to answer that together,  if they are not important enough for us maybe we can remove them from the guidelines. I personally think they are worth to be mentioned in the Testing section at least.